src/NHibernate/Async/Transaction/AdoTransaction.cs

```diff
@@ -0,0 +1,92 @@
+﻿//------------------------------------------------------------------------------
+// <auto-generated>
+//     This code was generated by AsyncGenerator.
+//
+//     Changes to this file may cause incorrect behavior and will be lost if
+//     the code is regenerated.
+// </auto-generated>
+//------------------------------------------------------------------------------
+
+
+using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Data.Common;
+
+using NHibernate.Engine;
+using NHibernate.Impl;
+
+namespace NHibernate.Transaction
+{
+	using System.Threading.Tasks;
+	using System.Threading;
+	/// <content>
+	/// Contains generated async methods
+	/// </content>
+	public partial class AdoTransaction : ITransaction
+	{
+
+		/// <summary>
+		/// Commits the <see cref="ITransaction"/> by flushing the <see cref="ISession"/>
+		/// and committing the <see cref="DbTransaction"/>.
+		/// </summary>
+		/// <param name="cancellationToken">A cancellation token that can be used to cancel the work</param>
+		/// <exception cref="TransactionException">
+		/// Thrown if there is any exception while trying to call <c>Commit()</c> on 
+		/// the underlying <see cref="DbTransaction"/>.
+		/// </exception>
+		public async Task CommitAsync(CancellationToken cancellationToken = default(CancellationToken))
```

Comment:
> I guess it will not be easy to add this information here

I will find a way to make [this](https://github.com/maca88/AsyncGenerator/issues/22) happen :)